app/src/main/java/com/android/unio/ui/event/EventCard.kt

```diff
@@ -0,0 +1,258 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalInspectionMode
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.core.net.toUri
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+import com.android.unio.R
+import com.android.unio.model.association.Association
+import com.android.unio.model.event.Event
+import com.android.unio.model.event.EventType
+import com.android.unio.model.event.EventViewModel
+import com.android.unio.model.event.MockEventRepository
+import com.android.unio.model.event.PreviewEventViewModel
+import com.android.unio.model.firestore.MockReferenceList
+import com.android.unio.model.map.Location
+import com.android.unio.model.user.MockUserRepository
+import com.android.unio.ui.theme.primaryContainerLight
+import com.android.unio.ui.theme.secondaryDark
+import com.android.unio.utils.EventUtils.addAlphaToColor
+import com.android.unio.utils.EventUtils.formatTimestamp
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+@Preview(showBackground = true)
+@Composable
+fun EventCardPreview() {
+  val sampleEvent =
+      Event(
+          title = "Sample Event",
+          image = "", // Empty string to test the placeholder
+          organisers = MockReferenceList<Association>(),
+          taggedAssociations = MockReferenceList<Association>(),
+          location = Location(0.0, 0.0, "aaaa"),
+          date = Timestamp(Date(2024 - 1900, 6, 20)),
+          catchyDescription = "This is a catchy description.",
+          types = listOf(EventType.TRIP))
+
+  val previewViewModel = PreviewEventViewModel(MockEventRepository(), MockUserRepository())
+  EventCard(event = sampleEvent, viewModel = previewViewModel)
+}
+
+@Composable
+fun EventCard(event: Event, viewModel: EventViewModel) {
+
+  var isSaved by remember { mutableStateOf(false) }
+
+  LaunchedEffect(event.uid) {
+    viewModel.isEventSavedForCurrentUser(event.uid) { saved -> isSaved = saved }
+  }
+
+  Column(
+      modifier =
+          Modifier.fillMaxWidth()
+              .padding(vertical = 8.dp)
+              .clickable(onClick = {})
```

Comment:
Perhaps it is better to simply not write the modifier if an empty callback function is put. However since this is a placeholder to add the logic later, consider adding a small comment to indicate it.

Reply:
That's sad because I loved doing that to remember to implement it haha but I think you are right !